app/services/alchemy/duplicate_element.rb

```diff
@@ -0,0 +1,51 @@
+# frozen_string_literal: true
+
+module Alchemy
+  class DuplicateElement
+    SKIPPED_ATTRIBUTES_ON_COPY = [
+      "cached_tag_list",
+      "created_at",
+      "creator_id",
+      "position",
+      "id",
+      "folded",
+      "updated_at",
+      "updater_id",
+    ].freeze
+
+    attr_reader :source_element, :repository
+
+    def initialize(source_element, repository: source_element.page_version.element_repository)
+      @source_element = source_element
+      @repository = repository
+    end
+
+    def call(differences = {})
+      attributes = source_element.attributes.with_indifferent_access
+        .except(*SKIPPED_ATTRIBUTES_ON_COPY)
+        .merge(differences)
+        .merge(
+          autogenerate_contents: false,
+          autogenerate_nested_elements: false,
+          tags: source_element.tags,
+        )
+
+      new_element = Element.create(attributes)
+
+      source_element.contents.map do |content|
+        Content.copy(content, element: new_element)
+      end
+
+      nested_elements = repository.children_of(source_element)
+      nested_elements.each.with_index do |nested_element, position|
```

Comment:
What it be helpful to do the same "disable acts-as-list and maintain the position ourselves" trick here as well?